src/wallet/mod.rs

```diff
@@ -983,6 +983,59 @@ impl Wallet {
         tracing::trace!("Trusted pending: {}", balance.trusted_pending);
         Ok(balance)
     }
+ 
+    pub async fn list_wallet_transactions(&self) -> Result<Vec<WalletTransaction>> {
+        let wallet = self.bitcoin_wallet.lock();
+        let transactions = wallet.transactions();
+        let mut txs = Vec::new();
+        
+        for tx in transactions {
```

Comment:
Would be nice to run all of these in parallell. Not that I know precisely how to do that, off the top of my head

Reply:
The call is basically instant. I'm all for optimizing performance, but not before it's a problem!